web-registry/src/components/organisms/BuyCreditsModal/BuyCreditsModal.tsx

```diff
@@ -130,15 +132,16 @@ const BuyCreditsModal: React.FC<React.PropsWithChildren<BuyCreditsModalProps>> =
       return errors;
     };
 
-    const allowedDenomsResponse =
-      useMarketplaceQuery<QueryAllowedDenomsResponse>({
-        query: 'allowedDenoms',
-        params: {},
-      });
+    const { data: allowedDenomsData } = useQuery(
+      getAllowedDenomQuery({
+        client: marketplaceClient,
+        enabled: !!marketplaceClient,
```

Comment:
It's important to use enabled for queries that require a client otherwise they will run once with an undefined client.

Reply:
do we have queries that would not require a client?

Reply:
Yes, coingecko or metadata ones for example.